opentelemetry-api/src/opentelemetry/attributes/__init__.py

```diff
@@ -179,23 +182,23 @@ def __setitem__(self, key, value):
                 ):
                     if not isinstance(self._dict, OrderedDict):
                         self._dict = OrderedDict(self._dict)
-                    self._dict.popitem(last=False)
+                    self._dict.popitem(last=False)  # type: ignore
                     self.dropped += 1
 
-                self._dict[key] = value
+                self._dict[key] = value  # type: ignore
 
-    def __delitem__(self, key):
-        if getattr(self, "_immutable", False):
+    def __delitem__(self, key: str) -> None:
+        if getattr(self, "_immutable", False):  # type: ignore
             raise TypeError
         with self._lock:
             del self._dict[key]
 
-    def __iter__(self):
+    def __iter__(self):  # type: ignore
         with self._lock:
-            return iter(self._dict.copy())
+            return iter(self._dict.copy())  # type: ignore
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self._dict)
 
-    def copy(self):
-        return self._dict.copy()
+    def copy(self):  # type: ignore
```

Comment:
Can this be inferred since we added type to `self.dict_` ?

Otherwise this would be:
```
Union[
            MutableMapping[str, types.AttributeValue],
            OrderedDict[str, types.AttributeValue],
        ]
```

Reply:
And it doesn't work